app/models/post.rb

```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def last_five_comments
+    Comment.where(post_id: id).last(5)
+  end
+
+  def update_post_counter(user)
+    user.update(posts_counter: Post.where(author_id: user.id).count)
+  end
```

Comment:
[OPTIONAL]

- I will strongly recommend that Students you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

e.g 

```ruby
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Nice tip